### ISSUE TITLE
Issue with getting next.js app running.

### ISSUE THREAD

**Issue by Reselence** [thebaselab/codeapp#238]
Firstly when running `npm i`, I always get WARNs that some tarball data seems to be corrupted, and that it's trying again, but nothing happens.

If I run it again I get `Cannot read property 'stdin' of undefined.`

Finally running `npm run dev` produces the following error: `npm ERR! command doesn't exist or is not supported: next dev`

Could anyone point me in the right direction please? I'm on the iPadOS 15 beta on the latest test flight.

Thank you.

**Comment by ZicklePop** [comment]
Same issue. When I try to just install Next I get an issue where Node is too old of a version so I'm wondering if #216 will help this too.

**Comment by Lukas-Batema** [comment]
> Same issue. When I try to just install Next I get an issue where Node is too old of a version so I'm wondering if #216 will help this too.

Next.js requires NODEJS version 12.**22**.0 (Reference: [Next.js Documentation](https://nextjs.org/docs)), whereas Code App runs NODEJS version 12.**19**.0 (Check settings in the Code App for reference). Which means @bummoblizard needs to update NODEJS in the app.

**Comment by truongvinht** [comment]
I think it also depends on which Next.js version is expected. play.js also runs an old version of next.js 9.3.5. But Code App doesn't. I was really hoping to switch to Code App. But it seems I still need to stick with play.js

**Comment by bummoblizard** [comment]
In that case, installing next.js 9.3.5 through npm should work. Next is not included in the app bundle.

> I think it also depends on which Next.js version is expected. play.js also runs an old version of next.js 9.3.5. But Code App doesn't. I was really hoping to switch to Code App. But it seems I still need to stick with play.js

**Comment by ZicklePop** [comment]
The current LTS version of Node is 16, so it feels like Code should keep up with that instead of downgrading.

**Comment by bummoblizard** [comment]
> The current LTS version of Node is 16, so it feels like Code should keep up with that instead of downgrading. Next is currently at version 12, so to go to 9 is a pretty big downgrade.

I agree. Currently we are on the same version as play.js. Unfortunately porting NodeJS to iOS presents a number of challenges as discussed here: https://github.com/thebaselab/codeapp/issues/328

Since codesandbox acquired play.js, there's some hope they will work on porting Node 12 to iOS and possibly open source it.
